.github/workflows/wheels.yml

```diff
@@ -195,12 +195,16 @@ jobs:
         name: dist
         path: dist/*.tar.gz
 
-  success:
-    permissions:
-      contents: none
+  pypi-publish:
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
     needs: [build, windows, sdist]
     runs-on: ubuntu-latest
-    name: Wheels Successful
+    name: Upload release to PyPI
+    environment:
+      name: pypi
```

Comment:
What is the effect of changing this?

Reply:
It's an arbitrary name of our choosing.

It must match the name we set in PyPI (see screenshot: https://github.com/python-pillow/Pillow/pull/7616#issuecomment-1855773030 (I can change this if needed)), and it will be the name of the environment we see in the GitHub UI.

Without an environment, the release will be published to PyPI from the CI workflow.

With an environment, a repo admin will first need to set one up for us. They will choose who can access it. The name above will be used here. With an env, the CI will first stage it in the env. Then someone with access can go and approve it and allow it to deploy.

It's an extra bit of security, but also an extra step in the release process. Without an env, anyone with write access to the repo and create tags, can publish. That's the five Pillow core devs. Currently, only the four of us listed at https://pypi.org/project/Pillow/ can `twine upload`. The env means we're essentially preserving the "maintainer" and "release manager" roles described at https://sethmlarson.dev/security-for-package-maintainers#platforms-and-roles.

Could be we don't need it? So far in other projects, I've not used envs because maintainers == release managers.

Anyway, `release-[test-]pypi` is perhaps a bit more descriptive than `[test-]pypi`, copying another [project](https://github.com/python-attrs/attrs/blob/main/.github/workflows/pypi-package.yml), but it's arbitrary and just a bit of bikeshedding.